### ISSUE TITLE
Give a name to a class

### ISSUE THREAD

**Issue by DamienCassou** [foretagsplatsen/klassified#45]
Currently, a class has no name. Even though that's not needed from a JS perspective, I think that's important for tools:

- showing a class hierarchy
- giving information about an object when debugging

It is possible to workaround this in tools in some cases:

- if a file just returns a class, we can give the class the same name as the file (without the file extension `.js`)
- if a file specifies several classes, it is possible that each class is set in a variable (e.g., `let myClass = superClass.subclass(...)`)

I don't see it as satisfactory enough.

**Comment by BenjaminVanRyseghem** [comment]
what about 

```
superclass.subclass("My New Awesome Class", (that, my) => { /* ... */ })
```

?

**Comment by DamienCassou** [comment]
perfect for me but it means we have to fix all classes. I can do it. @NicolasPetton?

**Comment by NicolasPetton** [comment]
We can do it, but let's this idea sink a bit in our brains first. I don't want to introduce a new API that we might change again in the future.

@DamienCassou I plan to work on similar things during the summer, so if you want you can make a branch if your desired changes, and we can integrate them in a few months if it still feels like the way to go.

**Comment by DamienCassou** [comment]
I agree we can wait a bit. Do we want to keep backward compatibility and keep anonymous classes around? I'm thinking about something ugly like: if first parameter is a string, it is considered the class name, otherwise it should be a function.
